test/util/CMakeLists.txt

```diff
@@ -0,0 +1,2 @@
+add_library(testUtil IdTableHelpers.cpp)
+qlever_target_link_libraries(testUtil engine)
```

Comment:
This shouldn't have to link against something as expensive as `engine`.
`IdTable` is header only anyway, same goes for `ValueId`.
If there are other requirements, maybe this is a good place to clean them up and split up the `cheap` and the `expensive` stuff.

Reply:
Ah, that would be the fault of `sortIdTableByJoinColumnInPlace`. It uses the engine function for sorting, that we also use in the actual engine.

I wrote it for the usage in benchmarks.

Reply:
> If there are other requirements, maybe this is a good place to clean them up and split up the `cheap` and the `expensive` stuff.

What do you mean with "clean up"? Splitting up `utilTest` into `utilTestCheap` and `utilTestExpensive` sounds rather dirty to me.
Both give you 0 idea about the difference in their content. And nobody will just know, which parts of the project are considered 'cheap'.

Reply:
Okay, 
Thanks for the clarification about the sorting. The argument could then be that we probably put the sorting in a different library, but I am not sure whether this should be done in this particular PR.

Reply:
How about splitting it up into `nonEngineTestUtil` and `engineTestUtil`? This should make things clearer.

Reply:
I think we can leave this for now and leave it for a different PR to systematically clean up these issues.
The best think that can happen to us in this respect are C++20 modules which still need some support in Compilers and CMake, but we are getting there.

Reply:
I've added a TODO about this in [b7a2c54](https://github.com/ad-freiburg/qlever/pull/1016/commits/b7a2c547f7e2c8ccfc0d44ba5357790cd668a84b).